### PR TITLE
Fix crash on use-after-free on macOS

### PR DESCRIPTION
**Pull request by kchibisov**
The code is inspired by glfw NSPasteboard clipboard loading.

Fixes alacritty/alacritty#6141.

### CODE REVIEW

**Comment by kchibisov** [comment]
@Lurk does this fixes your issue as well? Since without it alacritty segfaults right away after your patch in #33. 

For more see https://github.com/alacritty/alacritty/issues/6141.

**Review by chrisduerr — APPROVED** [review]
Makes sense to me.

**Comment by chrisduerr** [comment]
Relevant GLFW code: https://github.com/glfw/glfw/blob/d3ede7b6847b66cf30b067214b2b4b126d4c729b/src/cocoa_window.m#L1751

File: src/osx_clipboard.rs [hunk 1]
```diff
@@ -45,14 +45,23 @@ impl OSXClipboardContext {
 
 impl ClipboardProvider for OSXClipboardContext {
     fn get_contents(&mut self) -> Result<String> {
-        let nsstring: *mut NSString =
-            unsafe { msg_send![self.pasteboard, stringForType: NSPasteboardTypeString] };
-        if nsstring.is_null() {
-            Err("pasteboard#stringForType returned null".into())
-        } else {
-            let nsstring: Id<NSString> = unsafe { Id::from_retained_ptr(nsstring) };
```
Comment (madsmtm) [inline src/osx_clipboard.rs:53]: The issue in particular was using `Id::from_retained_ptr`; it should have been `Id::from_ptr`

**Review by madsmtm — APPROVED** [review]
Can attest that you now follow Cocoa's memory management rules properly (haven't tested it though)

**Comment by Lurk** [comment]
@kchibisov I just checked, and there is no leak.

**Pull request merged**